libminifi/test/TestBase.h

```diff
@@ -21,35 +21,104 @@
 #include <dirent.h>
 #include <cstdio>
 #include <cstdlib>
+#include <sstream>
 #include "ResourceClaim.h"
 #include "catch.hpp"
 #include <vector>
-#include "core/logging/LogAppenders.h"
 #include "core/logging/Logger.h"
 #include "core/Core.h"
 #include "properties/Configure.h"
+#include "properties/Properties.h"
+#include "core/logging/LoggerConfiguration.h"
+#include "spdlog/sinks/ostream_sink.h"
+#include "spdlog/sinks/dist_sink.h"
 
 class LogTestController {
  public:
-  LogTestController(const std::string level = "debug") {
-    logging::Logger::getLogger()->setLogLevel(level);
+  static LogTestController& getInstance() {
+   static LogTestController instance;
+   return instance;
   }
-
-  void enableDebug() {
-    logging::Logger::getLogger()->setLogLevel("debug");
+  
+  template<typename T>
+  void setDebug() {
+    setLevel<T>(spdlog::level::debug);
```

Comment:
@phrocker the LogTestController class is already coupled to spdlog (for construction of the sinks, etc) and was where I was trying to isolate spdlog in context of the tests.  I'd removed the parallel level enum logic we had in the code because it's only needed in tests as far as I can tell.